app/javascript/src/ollie-form.js

```diff
@@ -0,0 +1,58 @@
+export function bind() {
+  const element = document.querySelector("[data-role='ollie-form']")
+
+  if (!element) return
+
+  const usernameInput = element.querySelector("input[name='username']")
+  usernameInput.removeAndAddEventListener("input", event => setPupilPosition(event, element))
+  usernameInput.removeAndAddEventListener("focus", event => setPupilPosition(event, element))
+  usernameInput.removeAndAddEventListener("focus", () => setOllieBody(element, "base"))
+  usernameInput.removeAndAddEventListener("blur", () => resetPupilPosition(element))
+
+  const passwordInput = element.querySelector("input[type='password']")
+  passwordInput.removeAndAddEventListener("focus", () => setOllieBody(element, "eyes-closed"))
+  passwordInput.removeAndAddEventListener("blur", () => setOllieBody(element, "base"))
+
+  const rememberMeInput = element.querySelector("input[type='checkbox'][name='remember_me']")
+  rememberMeInput.removeAndAddEventListener("input", () => setOllieBody(element, "base"))
+}
+
+function setPupilPosition({ target }, element) {
+  const max = 32
+  const valueLength = Math.min(max, Math.max(0, target.value.length))
+  const xOffset = -1 + (valueLength / (max / 2)) * 2
+
+  const pupils = getOlliePart(element, "pupils")
+  pupils.style.transform = `translateX(${ xOffset }px) translateY(1px)`
```

Comment:
Maybe using percentages may give smaller steps. From my testing, 14% seems to be a good max percentage.